js/parsers/paragraphparser.js

```diff
@@ -7,37 +7,66 @@ const pptelement_1 = require("airppt-models/pptelement");
  * Parse the paragraph elements
  */
 class ParagraphParser {
-    static extractParagraphElements(textElement) {
-        if (!textElement || !textElement["a:r"]) {
+    static extractParagraphElements(paragraphs) {
+        if (!paragraphs || paragraphs.length === 0) {
             return null;
         }
-        let pptTextElement = {
-            text: this.ConcatenateParagraphLines(textElement["a:r"]) || "",
-            textCharacterProperties: this.determineTextProperties(checkobj_1.CheckValidObject(textElement, '["a:r"][0]["a:rPr"][0]')),
-            paragraphProperties: this.determineParagraphProperties(textElement)
-        };
-        return pptTextElement;
+        return paragraphs.map((paragraph) => {
+            const textElements = paragraph["a:r"] || [];
+            const content = textElements.map((txtElement) => {
+                return {
+                    text: txtElement["a:t"] || "",
+                    textCharacterProperties: this.determineTextProperties(checkobj_1.CheckValidObject(txtElement, '["a:rPr"][0]'))
+                };
+            });
+            return {
+                content: content,
+                paragraphProperties: this.determineParagraphProperties(paragraph)
+            };
+        });
     }
     /**a:rPr */
     static determineTextProperties(textProperties) {
         if (!textProperties) {
             return null;
         }
-        let textPropertiesElement = {
+        const textPropertiesElement = {
             size: checkobj_1.CheckValidObject(textProperties, '["$"].sz') || 1200,
             fontAttributes: this.determineFontAttributes(textProperties["$"]),
             font: checkobj_1.CheckValidObject(textProperties, '["a:latin"][0]["$"]["typeface"]') || "Helvetica",
             fillColor: colorparser_1.default.getTextColors(textProperties) || "000000"
         };
         return textPropertiesElement;
     }
+    /** Parse for italics, bold, underline & strike through*/
+    static determineFontAttributes(attributesList) {
+        let attributesArray = [];
+        if (!attributesList) {
+            return null;
+        }
+        Object.keys(attributesList).forEach((element) => {
+            if (element === "b" && attributesList[element] == 1) {
```

Comment:
can you define `b`, `u`, `i` etc. as enum or constants so they can be reused and references with clarity. If there is a constants file move them there, if not, then perhaps add one.

Reply:
changed enums values in the airppt-models and using them here now.
https://github.com/rabi92/airppt-models/commit/d260ef8f2695dd5061512ace5e1aa5f8d28454d2

Reply:
@rabi92 looks like that change will break output for @kryasaleg because now the parsed structure will show font attribute as 'b' instead of 'bold'. can you check?

Reply:
I was thinking of adding a new enum for the element attribute and using that and leaving the current one as is for the parsed output strucutre.

Reply:
this seemed more cleaner to me, and i don't think @kryasaleg has started working on this one yet. plus he will be using the same enum to compare so it won't make a difference.

Reply:
@rabi92 sure, I don't mind, if @kryasaleg is OK with this and keep as is.

Reply:
@rabi92 , I'm ok with it as I will reuse the same enum.